### PR TITLE
feat: strategy tooltip grouping and default

### PR DESCRIPTION
**Pull request by sjaanus**
image

### CODE REVIEW

File: frontend/src/component/feature/FeatureStrategy/FeatureStrategyMenu/FeatureStrategyMenuCards/FeatureStrategyMenuCards.tsx [hunk 1]
```diff
@@ -1,28 +1,89 @@
...
     }, [strategies]);
 
+    const { project } = useProject(projectId);
+
+    const strategy = project.environments.find(
+        env => env.environment === environmentId
+    )?.defaultStrategy;
+
+    const defaultStrategy: IStrategy | undefined = strategy
+        ? {
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureStrategy/FeatureStrategyMenu/FeatureStrategyMenuCards/FeatureStrategyMenuCards.tsx:48]: is this the only representation of the default strategy in the frontend code? or does it exist in other places and can get out of sync?

File: frontend/src/component/feature/FeatureStrategy/FeatureStrategyCreate/FeatureStrategyCreate.tsx [hunk 2]
```diff
@@ -30,14 +30,33 @@ import { useChangeRequestApi } from 'hooks/api/actions/useChangeRequestApi/useCh
...
+    );
+    const { project } = useProject(projectId);
+
+    const defaultStrategy = project.environments.find(
+        env => env.environment === environmentId
+    )?.defaultStrategy;
+
+    const { segments: allSegments } = useSegments();
+    const strategySegments =
+        allSegments?.filter(segment => {
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureStrategy/FeatureStrategyCreate/FeatureStrategyCreate.tsx:53]: (allSegments || []).filter()

File: frontend/src/component/feature/FeatureStrategy/FeatureStrategyCreate/FeatureStrategyCreate.tsx [hunk 3]
```diff
@@ -79,10 +98,12 @@ export const FeatureStrategyCreate = () => {
     }, [feature.name]);
 
     useEffect(() => {
-        if (strategyDefinition) {
+        if (useDefaultStrategy) {
+            setStrategy((defaultStrategy as any) || {});
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureStrategy/FeatureStrategyCreate/FeatureStrategyCreate.tsx:102]: can we do it without this any?

File: frontend/src/component/feature/FeatureStrategy/FeatureStrategyMenu/FeatureStrategyMenuCards/FeatureStrategyMenuCards.tsx [hunk 4]
```diff
@@ -1,28 +1,89 @@
...
     environmentId,
 }: IFeatureStrategyMenuCardsProps) => {
     const { strategies } = useStrategies();
+    const { uiConfig } = useUiConfig();
+
+    const preDefinedStrategies = useMemo(() => {
+        return strategies.filter(
+            strategy => !strategy.deprecated && !strategy.editable
+        );
+    }, [strategies]);
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureStrategy/FeatureStrategyMenu/FeatureStrategyMenuCards/FeatureStrategyMenuCards.tsx:33]: do we need those memos? also using objects in hooks as references never works correctly unless you have the exact same memory instance. It's comparison by === not by the actual values

**Review by kwasniew — APPROVED** [review]
Also maybe a test to increase confidence in this area of code would do

File: frontend/src/component/feature/FeatureStrategy/FeatureStrategyMenu/FeatureStrategyMenuCards/FeatureStrategyMenuCards.tsx (hunk 1 above)
Comment (sjaanus) [reply frontend/src/component/feature/FeatureStrategy/FeatureStrategyMenu/FeatureStrategyMenuCards/FeatureStrategyMenuCards.tsx:48]: This is only representation if IStrategy for default. yes

File: frontend/src/component/feature/FeatureStrategy/FeatureStrategyMenu/FeatureStrategyMenuCards/FeatureStrategyMenuCards.tsx (hunk 4 above)
Comment (sjaanus) [reply frontend/src/component/feature/FeatureStrategy/FeatureStrategyMenu/FeatureStrategyMenuCards/FeatureStrategyMenuCards.tsx:33]: Yes we can. Removed

**Pull request merged**